csharp/src/Ice/Reference.cs

```diff
@@ -750,9 +669,10 @@ public class FixedReference : Reference
 
             if (fixedConnection != null && fixedConnection != _fixedConnection)
             {
-                reference = Clone(reference);
+                reference = CloneIfSame(reference);
                 reference._fixedConnection = fixedConnection;
                 reference.Secure = fixedConnection.Endpoint.GetInfo().Secure();
+                reference._requestHandler = reference.CreateConnectionRequestHandler();
```

Comment:
if we keep this logic for cloning, it would be good to add a comment to make sure this is called last.

Reply:
Will do.